### PR TITLE
[KHP3-6461]  Added packages and interventions on the claims form

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Screenshots

claims

## Related Issue

*None.*

[[KHP3-6461] Added packages and interventions on the claims form - Jira (atlassian.net)](https://thepalladiumgroup.atlassian.net/browse/KHP3-6461)

## Other

*None.*

[KHP3-6461]: https://thepalladiumgroup.atlassian.net/browse/KHP3-6461?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Injiri** [comment]
@its-kios09  my take is  when one should pick one package  then one intervention that is within the package

**Comment by jecihjoy** [comment]
@its-kios09 please see below suggestions

1. Remove claim code
2. Change provider Name and autofill with current cashier information
3. Please share sample payload after submitting the form

**Comment by ojwanganto** [comment]
Thanks, @its-kios09. Did you use a mock benefits package payload?

**Comment by its-kios09** [comment]
> @its-kios09 my take is when one should pick one package then one intervention that is within the package

@Injiri what I have done is that I have use a mock-up data from this but I can update it

**Comment by its-kios09** [comment]
> [@jecihjoy](/jecihjoy)
> Contributor
> ###
> ** [jecihjoy](/jecihjoy) ** commented

@jecihjoy i have shared it on slack the one that has `concept_uuids` Ignore the first payload

**Comment by its-kios09** [comment]
> Thanks, @its-kios09. Did you use a mock benefits package payload?

@ojwanganto  yes i did 

below is the payload 
```

{
    "providedItems": {
        "597b7642-b817-4df1-acef-968284e93191": {
            "items": [
                {
                    "uuid": "294AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
                    "price": 100,
                    "quantity": 1
                }
            ],
            "explanation": "This",
            "justification": "this"
        },
        "94400c88-db51-4250-b02e-9a29b943686d": {
            "items": [
                {
                    "uuid": "19e1f7a9-52b4-4975-804d-5c74445be316",
                    "price": 50,
                    "quantity": 1
                }
            ],
            "explanation": "This",
            "justification": "this"
        },
        "c9927caf-f690-4aa7-b84d-123d59ca51bc": {
            "items": [
                {
                    "uuid": "294AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
                    "price": 100,
                    "quantity": 1
                }
            ],
            "explanation": "This",
            "justification": "this"
        },
        "bb812249-d4cd-4769-a388-f28a1ce74366": {
            "items": [
                {
                    "uuid": "432AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
                    "price": 500,
                    "quantity": 1
                }
            ],
            "explanation": "This",
            "justification": "this"
        }
    },
    "claimExplanation": "This",
    "claimJustification": "this",
    "startDate": "2024-05-29",
    "endDate": "2024-08-19",
    "location": "11161",
    "diagnoses": [
        "197AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
        "217AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
        "193AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
        "195AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
    ],
    "paidInFacility": true,
    "patient": "30fa18ca-5d3c-477d-b6d2-28975af0fd05",
    "visitType": "3371a4d4-f66f-4454-a86d-92c7b3da990c",
    "guaranteeId": "G12345687",
    "cashier": "e02c40e5-04e7-11e5-ae3c-a0b3cc4f922f",
    "use": "claim",
    "insurer": "SHA",
    "billNumber": "6f36c777-a1ae-4a07-a6ec-5ef5060a048f",
    "packages": [
        "SHA-002",
        "SHA-001"
    ],
    "interventions": [
        "SHA-001-01"
    ]
}



```

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 1]
```diff
@@ -31,18 +31,9 @@ type ClaimsFormProps = {
 };
 
 const ClaimsFormSchema = z.object({
-  claimCode: z.string().nonempty({ message: 'Claim code is required' }),
-  guaranteeId: z.string().nonempty({ message: 'Guarantee Id is required' }),
+  // guaranteeId: z.string().nonempty({ message: 'Guarantee Id is required' }),
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:34]:
```suggestion
```

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 2]
```diff
@@ -100,16 +95,17 @@ const ClaimsForm: React.FC<ClaimsFormProps> = ({ bill, selectedLineItems }) => {
     mode: 'all',
     resolver: zodResolver(ClaimsFormSchema),
     defaultValues: {
-      claimCode: '',
-      guaranteeId: '',
+      // guaranteeId: '',
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:98]:
```suggestion
    
```

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 3]
```diff
@@ -305,38 +364,37 @@ const ClaimsForm: React.FC<ClaimsFormProps> = ({ bill, selectedLineItems }) => {
...
+                    placeholder="Guarantee Id"
+                    labelText={t('guaranteeId', 'Guarantee Id')}
+                    invalid={!!errors.guaranteeId}
+                    invalidText={errors.guaranteeId?.message}
                   />
                 )}
               />
             </Layer>
-          </Column>
+          </Column> */}
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:397]: Just remove if it isn't going to be used

File: packages/esm-billing-app/src/invoice/invoice.component.tsx [hunk 4]
```diff
@@ -29,6 +37,7 @@ const Invoice: React.FC = () => {
   const { bill, isLoading: isLoadingBill, error } = useBill(billUuid);
   const [selectedLineItems, setSelectedLineItems] = useState([]);
   const componentRef = useRef<HTMLDivElement>(null);
+  const isProcessClaimsFormEnabled = useFeatureFlag('claims-form');
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/invoice/invoice.component.tsx:40]:
```suggestion
  const isProcessClaimsFormEnabled = useFeatureFlag('healthInformationExchange');
```

**Review by donaldkibet — COMMENTED** [review]
Thanks @its-kios09 Looks good to me.

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 1 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: sorted

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 2 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: same

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 3 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: sorted

File: packages/esm-billing-app/src/invoice/invoice.component.tsx (hunk 4 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/invoice/invoice.component.tsx]: updated

**Review by donaldkibet — APPROVED** [review]

**Comment by its-kios09** [comment]
@donaldkibet same here

**Comment by donaldkibet** [comment]
> @donaldkibet same here

Let work to merge this today ping @its-kios09

**Comment by its-kios09** [comment]
@donaldkibet Thanks, I have sorted out the conflicts issues

**Comment by its-kios09** [comment]
@donaldkibet fixed the conflicts

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 5]
```diff
@@ -141,12 +134,14 @@ const ClaimsForm: React.FC<ClaimsFormProps> = ({ bill, selectedLineItems }) => {
       paidInFacility: true,
       patient: patientUuid,
       visitType: visitUuid,
-      guaranteeId: data.guaranteeId,
-      providers: data.providerName.map((provider) => provider.id),
-      claimCode: data.claimCode,
+      guaranteeId: 'G-001',
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:137]: Should this be generated by the user

**Review by donaldkibet — APPROVED** [review]

**Pull request merged**